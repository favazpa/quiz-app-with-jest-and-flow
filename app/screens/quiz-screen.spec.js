//@flow

import 'react-native';
import React from 'react';
import QuizScreen from './quiz-screen';
import ShallowRenderer from 'react-test-renderer/shallow';

const renderer = new ShallowRenderer();

const renderApp = () => {
  renderer.render(<QuizScreen />);
  return renderer.getRenderOutput();
};

describe('QuizScreen', () => {
  test('renders correctly', () => {
    const result = renderApp();
    console.log('result',result)
    expect(result).toMatchSnapshot();
  });
});
