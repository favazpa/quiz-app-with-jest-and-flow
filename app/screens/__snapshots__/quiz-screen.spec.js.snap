// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuizScreen renders correctly 1`] = `
<View
  style={
    {
      "backgroundColor": "#1f2336",
      "height": "100%",
    }
  }
>
  <Header />
  <QuizNumber
    currentQuestionNumber={1}
    quizes={
      [
        {
          "category": "General Knowledge",
          "correct_answer": "Seven",
          "difficulty": "medium",
          "id": 1,
          "incorrect_answers": [
            "Nine",
            "Six",
            "Eight",
          ],
          "question": "How many days do we have in a week?",
          "type": "multiple",
        },
        {
          "category": "General Knowledge",
          "correct_answer": "365",
          "difficulty": "medium",
          "id": 2,
          "incorrect_answers": [
            "355",
            "346",
            "364",
          ],
          "question": " How many days are there in a year?",
          "type": "multiple",
        },
        {
          "category": "General Knowledge",
          "correct_answer": "7",
          "difficulty": "hard",
          "id": 3,
          "incorrect_answers": [
            "8",
            "5",
            "14",
          ],
          "question": "How many colors are there in a rainbow?",
          "type": "boolean",
        },
        {
          "category": "General Knowledge",
          "correct_answer": "Camel",
          "difficulty": "easy",
          "id": 4,
          "incorrect_answers": [
            "Lion",
            "Goat",
            "Eagle",
          ],
          "question": "Which animal is known as the "Ship of the Desert?"",
          "type": "multiple",
        },
        {
          "category": "General Knowledge",
          "correct_answer": "East",
          "difficulty": "medium",
          "id": 5,
          "incorrect_answers": [
            "West",
            "South",
            "North",
          ],
          "question": "In which direction does the sun rise?",
          "type": "multiple",
        },
      ]
    }
    style={
      {
        "color": "white",
        "fontSize": 21,
        "fontWeight": "bold",
        "marginHorizontal": 20,
        "marginTop": 50,
      }
    }
    totalQuizNumber={5}
  />
  <Divider
    style={
      {
        "backgroundColor": "white",
        "height": 1,
        "marginHorizontal": 20,
        "marginTop": 10,
      }
    }
  />
  <Question
    data={
      {
        "category": "General Knowledge",
        "correct_answer": "Seven",
        "difficulty": "medium",
        "id": 1,
        "incorrect_answers": [
          "Nine",
          "Six",
          "Eight",
        ],
        "question": "How many days do we have in a week?",
        "type": "multiple",
      }
    }
  />
  <Button
    onPress={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "orange",
        "borderRadius": 15,
        "bottom": 20,
        "height": 60,
        "justifyContent": "center",
        "marginHorizontal": 20,
        "position": "absolute",
        "width": "90%",
      }
    }
    tilteStyle={
      {
        "color": "white",
        "fontSize": 20,
        "fontWeight": "900",
      }
    }
    title="Submit"
  />
</View>
`;
